lib/Document/Renderer/DocumentRenderer.php

```diff
@@ -43,15 +43,15 @@ class DocumentRenderer implements DocumentRendererInterface
 
     private EventDispatcherInterface $eventDispatcher;
 
-    private LocaleService $localeService;
+    private LocaleServiceInterface $localeService;
 
     public function __construct(
         RequestHelper $requestHelper,
         ActionRenderer $actionRenderer,
         FragmentRendererInterface $fragmentRenderer,
         DocumentRouteHandler $documentRouteHandler,
         EventDispatcherInterface $eventDispatcher,
-        LocaleService $localeService
+        LocaleServiceInterface $localeService
```

Comment:
@blankse Isn't this a BC-break, when someone extend from this class?

Reply:
No, it's not, because it's a constructor: https://3v4l.org/vZnkL
If you change the method name in the example to something else, you'll get an exception (=BC break) but constructors have different rules (otherwise it wouldn't be possible to have additional dependencies in subclasses, for example).

Reply:
@robertSt7 @jdreesen Its not a BC break. I tested it also with the old type hint to overwrite it. It works.

I noticed that the DocumentRender is also used directly. There is also a DocumentRenderInterface which should be used instead...

Reply:
I created a additional PR for this for 11.2. Because we need a deprecation there: https://github.com/pimcore/pimcore/pull/16267

Reply:
I created also a PR to deprecate the LocaleService aliases in 11.2 https://github.com/pimcore/pimcore/pull/16268

Reply:
@jdreesen  & @blankse Thanks a lot for clarification